samples/augment-reality-to-show-tabletop-scene/src/main/java/com/esri/arcgismaps/sample/augmentrealitytoshowtabletopscene/screens/AugmentRealityToShowTabletopSceneScreen.kt

```diff
@@ -35,30 +47,75 @@ import com.esri.arcgismaps.sample.sampleslib.components.SampleTopAppBar
-                    x = -75.16996728256345,
-                    y = 39.95787000283599,
-                    spatialReference = SpatialReference.wgs84()
+                    x = -75.16996728256345, y = 39.95787000283599, spatialReference = SpatialReference.wgs84()
                 ),
                 translationFactor = 800.0,
+                modifier = Modifier
+                    .fillMaxSize()
+                    .padding(paddingValues),
                 clippingDistance = 800.0,
+                onInitializationStatusChanged = { statusChanged ->
+                    initializationStatus = statusChanged
+                },
             )
         }
+    })
 
-        sceneViewModel.messageDialogVM.apply {
-            if (dialogStatus) {
-                MessageDialog(
-                    title = messageTitle, description = messageDescription, onDismissRequest = ::dismissDialog
-                )
+    // Show an overlay with instructions or progress indicator based on the initialization status
+    when (val status = initializationStatus) {
+        is TableTopSceneViewStatus.Initializing -> ShowInstruction(text = stringResource(R.string.initializing_overlay))
+        is TableTopSceneViewStatus.DetectingPlanes -> ShowInstruction(text = stringResource(R.string.detect_planes_overlay))
+        is TableTopSceneViewStatus.Initialized -> {
+            sceneViewModel.scene?.let {
+                when (val sceneLoadStatus = it.loadStatus.collectAsStateWithLifecycle().value) {
+                    // Tell the user to tap the screen if the scene has not started loading
+                    is LoadStatus.NotLoaded -> ShowInstruction(text = stringResource(R.string.tap_scene_overlay))
+                    // The scene may take a while to load, so show a progress indicator
+                    is LoadStatus.Loading -> {
+                        Column(
+                            modifier = Modifier.fillMaxSize(),
+                            verticalArrangement = Arrangement.Center,
+                            horizontalAlignment = Alignment.CenterHorizontally
+                        ) {
+                            CircularProgressIndicator()
+                        }
+                    }
+                    // Show an error message if the scene failed to load
+                    is LoadStatus.FailedToLoad -> ShowInstruction(
+                        text = stringResource(
+                            R.string.failed_to_load_scene, sceneLoadStatus.error
+                        )
+                    )
+                    is LoadStatus.Loaded -> {} // Do nothing
+                }
             }
         }
-    })
+        is TableTopSceneViewStatus.FailedToInitialize -> {
+            ShowInstruction(
+                text = stringResource(R.string.failed_to_initialize_overlay, status.error.message ?: status.error)
+            )
+        }
+    }
+}
+
+/**
+ * Displays the provided [text] on top of a half-transparent gray background.
```

Comment:
I don't see the transparent gray background in the app. It is a little hard to read without it, but up to you if you want to add that back in.

```suggestion
 * Displays the provided [text] in the middle of the screen.
```

Reply:
I dropped the transparent background because it went over the top app bar and didn't want to refactor just to account for that